### ISSUE TITLE
Using DADA2 with Low Biomass Samples

### ISSUE THREAD

**Issue by liisaveerus** [benjjneb/dada2#796]
Thanks for maintaining such a well-explained pipeline!

I have a couple of questions for you when using extremely low biomass samples. I work with bacteria harvested from reproductive organs of an avian host. Needless to say, the more internal we go, the less bacteria we detect in the reproductive tract.

I am currently using the 1.12 version with the following modifications:

- I filter, but do not trim reads as we run a 2*150 bp sequencing on a 291 bp insert (515F-806R primers), i.e. the overlap between forward and reverse reads is already lower than recommended by your pipeline (9 bp vs 12 bp).
- I run the error learning over a randomised set of samples to allow for differences between tissue and lavage samples (tested it in many different ways and it is surprisingly repeatable, no matter what samples are picked).
- Then, I run sample inference with pseudo-pooling to account for rare sequence variants across samples. 
- Finally, I merge the pairs with minOverlap = 9 to actually have any merging pairs. This is followed by a chimera check. 

Sending you a snippet of read tracking per bioinformatic step.

![Snippet](https://user-images.githubusercontent.com/52078155/59931027-012a2a80-943c-11e9-89c6-0ac923f0b59b.png)

You can see that the main drop comes from the merging step. I assume this is mostly due to the short overlap in our workflow + the general low quality at the end of the reads.

(1) Is there anything you would recommend to avoid losing such chunks of data? Unfortunately, repeated sequencing with 2*250 bp is impossible. Also, loosening the quality of mergePairs seems like a bad idea, considering my overlap is already so short. 

(2) I also lose 40% of ASVs (both in number and abundance) during the chimera check. Do you know why that may be? Are these potential contaminants? What would you recommend changing for the chimera check step (I run it without any modifications)? NB! All adapters have been removed!

(3) I have also tried to play around with decontam, as I have taken several different controls through the methods pipeline (extraction kit, PCR etc). I unfortunately do not have qPCR quantifications for the samples and Qubit quantifications cannot be trusted for concentration due to host DNA contamination. I am thus reliant on prevalence data that cannot detect any contaminants due to the high contaminant level in the extremely low biomass samples. What would you recommend in this case? I was thinking of removing all control-associated ASVs from my data set. Yes, that leads to removing some biologically-relevant data, but at least there are no contaminants in my data set.

I emailed you one sample (the first one in the snippet), if you would like to have a look. Thank you so much for your time!

**Comment by benjjneb** [comment]
> (1) Is there anything you would recommend to avoid losing such chunks of data? Unfortunately, repeated sequencing with 2*250 bp is impossible. Also, loosening the quality of mergePairs seems like a bad idea, considering my overlap is already so short.

You absolutely must reduce the `minOverlap` parameter in `mergePairs`. You only have 9 nts on average of overlap, and `mergePairs` requires 12 by default. That can be reduced, and has to be in your case. I would recomend reducing it as much as makes sense, probably to `minOverlap=5`, to account for biological length variation.

In a perfect world this isn't ideal, it would be better to have more overlap in the reads, but given your sequencing setup this is the best thing to do now.

> (2) I also lose 40% of ASVs (both in number and abundance) during the chimera check. Do you know why that may be?

I think you have primers on your reads. If you have 291 nt v4 reads, you do. You need to remove them. You can use `filterAndTrim(..., trimLeft=c(FWDPRIMERLEN, REVPRIMERLEN))` to do so. This is necessary.

> (3) I have also tried to play around with decontam, as I have taken several different controls through the methods pipeline (extraction kit, PCR etc). I unfortunately do not have qPCR quantifications for the samples and Qubit quantifications cannot be trusted for concentration due to host DNA contamination. I am thus reliant on prevalence data that cannot detect any contaminants due to the high contaminant level in the extremely low biomass samples. What would you recommend in this case? I was thinking of removing all control-associated ASVs from my data set. Yes, that leads to removing some biologically-relevant data, but at least there are no contaminants in my data set.

I recommend revisiting this once the processing issues above are fixed.

**Comment by liisaveerus** [comment]
Hi Benjamin,

Thanks so much for your pointers! You are completely right, there were first round primers still attached.

I changed my code in the following way:

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(150,150),
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                     **trimLeft=c(19, 20)**,
                     compress=TRUE, multithread=FALSE)

mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE, **minOverlap = 5**)

and this significantly helped with the merging and chimera check:

![Rates](https://user-images.githubusercontent.com/52078155/60594409-91a62a80-9dad-11e9-9cb6-da500ee7037d.png)

Now, I lost 79% of the merged sequence variants, but only 7% of the total reads during chimera check.

For other beginners, I really recommend looking into whether primers need to be removed; the assigning of bacterial taxonomies was much smoother and more precise too!

I also ran decontam with the new data set and posted a follow-up question in the relevant forum.

For anyone interested: https://github.com/benjjneb/decontam/issues/50
